pkg/pool-hooks/contracts/StableSurgeHookExample.sol

```diff
@@ -0,0 +1,235 @@
+    uint256 private _surgeCoefficient;
+
+    /**
+     * @notice A new `StableSurgeHookExample` contract has been registered successfully.
+     * @dev If the registration fails the call will revert, so there will be no event.
+     * @param hooksContract This contract
+     * @param factory The factory (must be the allowed factory, or the call will revert)
+     * @param pool The pool on which the hook was registered
+     */
+    event StableSurgeHookExampleRegistered(
+        address indexed hooksContract,
+        address indexed factory,
+        address indexed pool
+    );
+
+    /**
+     * @notice The threshold has been changed in a `StableSurgeHookExample` contract.
+     * @dev Note, the initial threshold is set on deployment and an event is emitted.
+     * @param hooksContract This contract
+     * @param threshold The new threshold
+     */
+    event ThresholdChanged(address indexed hooksContract, uint256 indexed threshold);
+
+    /**
+     * @notice The surgeCoefficient has been changed in a `StableSurgeHookExample` contract.
+     * @dev Note, the initial surgeCoefficient is set on deployment and an event is emitted.
+     * @param hooksContract This contract
+     * @param surgeCoefficient The new surgeCoefficient
+     */
+    event SurgeCoefficientChanged(address indexed hooksContract, uint256 indexed surgeCoefficient);
+
+    constructor(
+        IVault vault,
+        address allowedFactory,
+        uint256 threshold,
+        uint256 surgeCoefficient
+    ) VaultGuard(vault) Ownable(msg.sender) {
+        _allowedFactory = allowedFactory;
+        _threshold = threshold;
+        _surgeCoefficient = surgeCoefficient;
+        emit ThresholdChanged(address(this), threshold);
+        emit SurgeCoefficientChanged(address(this), surgeCoefficient);
+    }
+
+    /// @inheritdoc IHooks
+    function getHookFlags() public pure override returns (HookFlags memory hookFlags) {
+        hookFlags.shouldCallComputeDynamicSwapFee = true;
+    }
+
+    /// @inheritdoc IHooks
+    function onRegister(
+        address factory,
+        address pool,
+        TokenConfig[] memory,
+        LiquidityManagement calldata
+    ) public override onlyVault returns (bool) {
+        // This hook implements a restrictive approach, where we check if the factory is an allowed factory and if
+        // the pool was created by the allowed factory.
+        emit StableSurgeHookExampleRegistered(address(this), factory, pool);
+
+        return factory == _allowedFactory && IBasePoolFactory(factory).isPoolFromFactory(pool);
+    }
+
+    /// @inheritdoc IHooks
+    function onComputeDynamicSwapFeePercentage(
+        PoolSwapParams calldata params,
+        address pool,
+        uint256 staticSwapFeePercentage
+    ) public view override onlyVault returns (bool, uint256) {
+        uint256 amp;
+        (amp, , ) = IStablePool(pool).getAmplificationParameter();
```

Comment:
Does this all still work if the amp factor's changing?

Reply:
From my understanding `getAmplificationParameter` will return the instantaneous value of the amp which is then used in the calculations so I believe this works.